test/core/renderControllerTests.ts

```diff
@@ -1,8 +1,80 @@
 ///<reference path="../testReference.ts" />
 
 describe("RenderController", () => {
-  // HACKHACK: #2083
-  it.skip("Components whose render() is triggered by another Component's render() will be drawn", () => {
+  let SVG_WIDTH = 400;
+  let SVG_HEIGHT = 300;
+
+  describe("configuring the render policy", () => {
+    let oldWarn: (msg: string) => void;
+    let warned: boolean;
+
+    before(() => {
+      oldWarn = Plottable.Utils.Window.warn;
+      Plottable.Utils.Window.warn = () => {
+        warned = true;
+      };
+    });
+
+    beforeEach(() => {
+      warned = false;
+    });
+
+    after(() => {
+      Plottable.RenderController.renderPolicy(Plottable.RenderController.Policy.IMMEDIATE);
+      Plottable.Utils.Window.warn = oldWarn;
+    });
+
+    it("can set a render policy", () => {
+      let renderPolicy = Plottable.RenderController.Policy.TIMEOUT;
+      Plottable.RenderController.renderPolicy(renderPolicy);
+
+      assert.strictEqual(Object.getPrototypeOf(Plottable.RenderController.renderPolicy()),
+        Plottable.RenderPolicies.Timeout.prototype, "render policy is of the same type");
+    });
+
+    it("throws a warning for unrecognized render policies", () => {
+      let unrecognizedRenderPolicy = "foo";
+      Plottable.RenderController.renderPolicy(unrecognizedRenderPolicy);
+
+      assert.isTrue(warned, "warning sent for unrecognized render policy");
+    });
+  });
+
+  it("can queue a component to render", () => {
+    let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+    let component = new Plottable.Component();
+    let rendered = false;
+    component.renderImmediately = () => {
+      rendered = true;
+      return component;
+    };
+    component.anchor(svg);
+    Plottable.RenderController.registerToRender(component);
+    assert.isTrue(rendered, "component has rendered");
+    component.destroy();
+    svg.remove();
+  });
+
+  it("can queue a component to undergo layout computation and render", () => {
+    let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+    let component = new Plottable.Component();
+    let rendered = false;
+    component.renderImmediately = () => {
```

Comment:
Less performant, but probably more safe.  I'm thinking of making an element and then checking that the element exists